src/backend/utils/misc/guc.c

```diff
@@ -13075,4 +13083,51 @@ guc_set_stack_value(struct config_generic *gconf, config_var_value *val)
 	set_stack_value(gconf,  val);
 }
 
+static char*
+transaction_isolation_to_string(int v)
+{
+	switch (v)
+	{
+		case XACT_SERIALIZABLE:
+			return "serializable";
+		case XACT_REPEATABLE_READ:
+			return "repeatable read";
+		case XACT_READ_COMMITTED:
+			return "read committed";
+		case XACT_READ_UNCOMMITTED:
+			return "read uncommitted";
+	}
+	return "(unrecognized)";
+}
+
+static bool
+check_default_transaction_isolation(int *newval, void **extra, GucSource source)
+{
+	if (guc_newval_hook)
+	{	
+		int			newXactIsoLevel = *newval;
+		const char*		ignore_isolation;
+
+		if ( newXactIsoLevel != XactIsoLevel && IsTransactionState() )
+		{
+			if( FirstSnapshotSet || IsSubTransaction() || (newXactIsoLevel == XACT_SERIALIZABLE && RecoveryInProgress()))
+			{
+				ignore_isolation = GetConfigOption("babelfishpg_tsql.escape_hatch_set_transaction_isolation_level", false, false);
+
+				if(ignore_isolation && !strcmp(ignore_isolation, "ignore"))
+				{
+					*newval = XactIsoLevel;
```

Comment:
if we put everything in a single hook, we can simply skip setting transaction isolation level